internal/sync/syncer.go

```diff
@@ -355,13 +366,31 @@ func (s *Syncer) compareState(gitSync *v1alpha1.GitSync, targetObjs []*unstructu
 	return reconciliationResult, diffResults, nil
 }
 
-func applyAnnotation(manifests []*unstructured.Unstructured, gitSyncName string) ([]*unstructured.Unstructured, error) {
+// getResourceOperations will return the kubectl implementation of the ResourceOperations
+// interface that provides functionality to manage kubernetes resources. Returns a
+// cleanup function that must be called to remove the generated kube config for this
+// server.
+func (s *Syncer) getResourceOperations() (kube.ResourceOperations, func(), error) {
+	clusterCache, err := s.stateCache.GetClusterCache()
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting cluster cache: %w", err)
+	}
+
+	ops, cleanup, err := s.kubectl.ManageResources(s.config, clusterCache.GetOpenAPISchema())
+	if err != nil {
+		return nil, nil, fmt.Errorf("error creating kubectl ResourceOperations: %w", err)
+	}
+	return ops, cleanup, nil
+}
+
+func applyAnnotationAndNamespace(manifests []*unstructured.Unstructured, gitSyncName, destinationNamespace string) ([]*unstructured.Unstructured, error) {
 	uns := make([]*unstructured.Unstructured, 0)
 	for _, m := range manifests {
 		err := kubernetes.SetGitSyncInstanceAnnotation(m, common.AnnotationKeyGitSyncInstance, gitSyncName)
 		if err != nil {
 			return nil, err
 		}
+		m.SetNamespace(destinationNamespace)
```

Comment:
Is this just for extra safety? Or we have a bug without it? Shouldn't when we set the namespace in syncing taking care of it?

Reply:
After enabling server side diff, without knowing the namespace it was not able to find the resource in cluster for diff. that's why we need to set it upfront